// This file was generated with the Protobuf generator tool.
// Dataset ID    : 31
// Dataset Name  : credit-g;
// Dataset URL   : https://www.openml.org/data/v1/download/31/credit-g.arff
// Num. Columns  : 21
// Num. Rows     : 1000
// Target Feature: class

// Beginning of Description of Dataset:
// **Author**: Dr. Hans Hofmann  
// **Source**: [UCI](https://archive.ics.uci.edu/ml/datasets/statlog+(german+credit+data)) - 1994    
// **Please cite**: [UCI](https://archive.ics.uci.edu/ml/citation_policy.html)
// **German Credit dataset**  
// This dataset classifies people described by a set of attributes as good or bad credit risks.
// This dataset comes with a cost matrix: 
// ``` 
// Good  Bad (predicted)  
// Good   0    1   (actual)  
// Bad    5    0  
// ```
// It is worse to class a customer as good when they are bad (5), than it is to class a customer as bad when they are good (1).  
// ### Attribute description  
// 1. Status of existing checking account, in Deutsche Mark.  
// 2. Duration in months  
// 3. Credit history (credits taken, paid back duly, delays, critical accounts)  
// 4. Purpose of the credit (car, tel
//

syntax = "proto3";

message Empty {
}

message Features {
	string   Checking_status                = 1;
	uint32   Duration                       = 2;
	string   Credit_history                 = 3;
	string   Purpose                        = 4;
	double   Credit_amount                  = 5;
	string   Savings_status                 = 6;
	string   Employment                     = 7;
	uint32   Installment_commitment         = 8;
	string   Personal_status                = 9;
	string   Other_parties                  = 10;
	uint32   Residence_since                = 11;
	string   Property_magnitude             = 12;
	uint32   Age                            = 13;
	string   Other_payment_plans            = 14;
	string   Housing                        = 15;
	uint32   Existing_credits               = 16;
	string   Job                            = 17;
	uint32   Num_dependents                 = 18;
	string   Own_telephone                  = 19;
	string   Foreign_worker                 = 20;
	string   Class                          = 21;
}

service get_next_row {
	 rpc get_next_row(Empty) returns(Features);
}

/*
//This code section can be directly used
//at the end of gen_next_row in the server.py file
//for OpenML dataset nr. 31
        response.Checking_status                = row[0]
        response.Duration                       = numpy.uint32(row[1])
        response.Credit_history                 = row[2]
        response.Purpose                        = row[3]
        response.Credit_amount                  = row[4]
        response.Savings_status                 = row[5]
        response.Employment                     = row[6]
        response.Installment_commitment         = numpy.uint32(row[7])
        response.Personal_status                = row[8]
        response.Other_parties                  = row[9]
        response.Residence_since                = numpy.uint32(row[10])
        response.Property_magnitude             = row[11]
        response.Age                            = numpy.uint32(row[12])
        response.Other_payment_plans            = row[13]
        response.Housing                        = row[14]
        response.Existing_credits               = numpy.uint32(row[15])
        response.Job                            = row[16]
        response.Num_dependents                 = numpy.uint32(row[17])
        response.Own_telephone                  = row[18]
        response.Foreign_worker                 = row[19]
        response.Class                          = row[20]
*/

